src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/Food.java

```diff
@@ -0,0 +1,5 @@
+package com.epam.training.designpatterns.fastfoodrestaurant;
+
+public interface Food {
+    double effect(double happines);
```

Comment:
Nagyon jó. Bemegy a régi érték, kijön az új, módosított érték. A mögöttes logikát pedig elrejti az étel. Akár az aktuális `Client` példányt is megkaphatná ez a metódus és módosíthatná a `hapiness` értékét.